src/Screen/Screen.php

```diff
@@ -236,7 +236,11 @@ private function bind(int $key, ReflectionParameter $parameter, array $httpQuery
         $object = resolve($class);
 
         if ($original !== null && is_a($object, UrlRoutable::class)) {
-            return $object->resolveRouteBinding($original);
+            $object_to_return = $object->resolveRouteBinding($original);
+            if (key_exists($parameter->getName(),request()->route()->parameters)){
```

Comment:
Are you sure we need this check? Don't we do the same a few levels higher.

Reply:
Well in order to replicate normal controllers behavior on binding:

   When you pass a typehint FormRequest object on a controller method as param, the request only has the value of a resource/model in the route attribute,
   
   But if you pass the model, as typehint object and also a formrequest in the controller method heading, then the route attribute value is replaced by the model with data from the db that match the old route attribute value as id.
   
   I don't see that in upper levels.

Reply:
off course the route  attribute name, and the controller method typedhint object name as parameter must match to do the route attribute replacement

Reply:
You are right, Check is not needed